core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -440,8 +440,12 @@ object StorageTool extends Logging {
         "Use --ignore-formatted to ignore this directory and format the others.")
     }
     if (!copier.errorLogDirs().isEmpty) {
-      val firstLogDir = copier.errorLogDirs().iterator().next()
-      throw new TerseFailure(s"I/O error trying to read log directory $firstLogDir.")
+      copier.errorLogDirs().forEach(errorLogDir => {
+        stream.println(s"I/O error trying to read log directory $errorLogDir. Ignoring...")
+      })
+      if(metaPropertiesEnsemble.emptyLogDirs().isEmpty) {
+        throw new TerseFailure("No available log directories to format.")
```

Comment:
Do we need to check `logDirProps` before throwing this exception?

Reply:
Discussed below. I've now added the additional check to see if `logDirProps` is empty, and only throw this failure when this is also true. This is to prevent the format storage script from failing when there is at least one directory already formatted, and the only other directories are unavailable, when it is ran with `--ignore-formatted`.